qt/python/mantidqt/plots/pyplot.py

```diff
@@ -0,0 +1,58 @@
+def plot(axes, workspace, *args, **kwargs): #specNum=None, wkspIndex=None, distribution=None, *args, **kwargs):
```

Comment:
This looks really nice.

My only concern here is that as we've labelled the module `pyplot` then I think it should be possible to get a plot with a single line as in `mpl.pyplot`. I think users will quickly get irritated if they always have to create the figures and axes themselves. Can we make them optional and use `matplotlib.pyplot.gca()` to create one if we're not given on?

Reply:
I don't want to encourage users to use the stateful matplotlib API. It creates all sorts of issues. I'd rather rename this module to better communicate what it is.

Reply:
How about `mantidqt.plots.workspace`?

I see your point for scripts - it's much easier to deal with manipulating existing plots this way.
I think we will rapidly be asked for one liners for the common plot types though. Especially if you want to do something quickly from the IPython terminal.

The other question is what about the custom toolbars for 1D & 2D plots that we had designed - how would we deal with those?